lib/tls_syntax/src/tls_syntax.cpp

```diff
@@ -3,22 +3,46 @@
 // NOLINTNEXTLINE(llvmlibc-implementation-in-namespace)
 namespace tls {
 
+void
+ostream::check_remaining(size_t length)
+{
+  if (length > _buffer.size()) {
+    throw WriteError("Write size exceeds available size");
+  }
+}
+
 void
 ostream::write_raw(const std::vector<uint8_t>& bytes)
 {
-  // Not sure what the default argument is here
-  // NOLINTNEXTLINE(fuchsia-default-arguments)
-  _buffer.insert(_buffer.end(), bytes.begin(), bytes.end());
+  check_remaining(bytes.size());
+
+  // XXX(RLB) This is a hack around the const-ness of string_view
+  auto* data = const_cast<uint8_t*>(_buffer.data());
```

Comment:
Dirty hack. Sounds like some kind of monothonic buffer is needed.

Reply:
What we really want here is [`std::span`](https://en.cppreference.com/w/cpp/container/span).  The alternatives are:

* Take on a C++20 dependency
* Bring in a GSL library or other similar polyfill as a dependency
* Include an implementation of the buffer type here

... none of which seem more appealing than this small hack.